src/sentry/migrations/0549_flat_file_size_cleanup.py

```diff
@@ -0,0 +1,74 @@
+# Generated by Django 3.2.20 on 2023-09-08 09:41
+
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0548_add_is_unclaimed_boolean_to_user"),
+    ]
+
+    operations = [
+        # Add new `size` field
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(
+                    """
+                    ALTER TABLE "sentry_artifactbundleflatfileindex" ADD COLUMN "size" integer NOT NULL DEFAULT 0;
+                    """,
+                    reverse_sql="""
+                    ALTER TABLE "sentry_artifactbundleflatfileindex" DROP COLUMN "size";
+                    """,
+                    hints={"tables": ["sentry_artifactbundleflatfileindex"]},
+                ),
+            ],
+            state_operations=[
+                migrations.AddField(
+                    model_name="artifactbundleflatfileindex",
+                    name="size",
+                    field=sentry.db.models.fields.bounded.BoundedPositiveIntegerField(default=0),
+                ),
+            ],
+        ),
+        # Give fields a default so they can be removed in followup migrations
+        migrations.AlterField(
+            model_name="artifactbundleindex",
+            name="release_name",
+            field=models.CharField(default="", max_length=250),
+        ),
+        migrations.AlterField(
+            model_name="debugidartifactbundle",
+            name="source_file_type",
+            field=models.IntegerField(default=2),
+        ),
```

Comment:
The SQL dump for this migration isn't showing any schema changes for these default values, as django likes to manage default values in the ORM. You would need to use `SeparateDatabaseAndState` to ensure DDL is applied. You can also make the fields nullable which will also generate DDL.

Reply:
jesus… ofc the existing `default`s of the columns are not in the DDL either but just in django land 🤦🏻 

So I changed this yet again to first set the columns as nullable, hopefully that will finally do the right thing.